### PR TITLE
Fixes imports of published html files

### PR DESCRIPTION
**Pull request by jonuy**
#### What's this PR do?

Fixes the importing of published html files that have our custom passages in them.
#### Where should the reviewer start?
- `importFile()` - This previously only expected and supported data from normal Passage models. This instead passes `$passage` data onto `createPassage()` to figure out what type of passage it's dealing with and creating the appropriate object to be turned into a Backbone.Model with `allPassages.create`.
- `createPassage()` - Some of the old `importFile` logic got moved down here. But the primary source of magic in this function is with `buildPassage()`. It matches passage types and when one is found, iterates through that object's keys and uses that to transfer data between the $passage jquery object to the resulting object that'll ultimately be used to create a Backbone.Model.
#### How should this be manually tested?

I used an previously created game that had StoryConfig, Passage, and PassageDS passages. Published it to a file. Then imported it again with that file. Verified that all the data was there in the correct type of passage.

I also tested a regular Twine game to make sure that import would work.
#### What are the relevant tickets?
#32 - this might close the issue, but I'll want to test again with the endgame passages.

### CODE REVIEW

**Comment by jonuy** [comment]
@tongxiang It does seem to work with your endgame changes! So yea, this should just close out #32.

File: js/app.js [hunk 1]
```diff
@@ -206,29 +207,26 @@ var TwineApp = Backbone.Marionette.Application.extend(
...
-          left: parseInt(posBits[0]),
-          top: parseInt(posBits[1])
-        }, { wait: true }); 
-
-        if (id == startPassageId)
-          story.save({ startPassage: passage.id });
-      });
+        var passageObj = createPassage($passage, story.id);
+
+        var passageModel = allPassages.create(passageObj, { wait: true });
```
Comment (tongxiang) [inline js/app.js:221]: Still confused what's going on this line. So this essentially is a backbone call for: 

`Passage.collection.all().create(passageObj, {wait: true})`

My questions: 
1. Why is the `.all()` call necessary? Can't find an example of that online. 
2. How is Passage smart enough to know which type of passage to create?

**Comment by tongxiang** [comment]
it all works! :+1:

File: js/app.js (hunk 1 above)
Comment (jonuy) [inline js/app.js:221]: 1) Yea, this was me not trying to rock the boat too much. Previously we saw in developing importsms.js that we could do something like...

```
var passageCollection = new PassageCollection();
passageCollection.create(passageObj, {wait: true}); 
```

... and that would work. So we could technically do that here too. I just kept it the way it was to not change more than I needed to. But I'm sort of ambivalent now to changing/fixing how that's done.

2) If you look in `PassageCollection.model` you'll see it creating the different kinds of models based on the type. The [docs](http://backbonejs.org/#Collection-model) tell us that this is used in `Collection.add`, `.create`, and `.reset`.

Comment (tongxiang) [inline js/app.js:221]: 1. Ahhh okay that makes sense. I'm fine with keeping it with what it is. 
2. Oh right re: `PassageCollection.model`! Gotcha, backbone.

**Pull request merged**